action.yml

```diff
@@ -191,6 +200,45 @@ runs:
       set +x
     shell: bash
 
+  - name: Log the next step action
+    run: echo ▷ Figuring out change detection and coverage
+    shell: bash
+  - name: Determine change detection and coverage
+    id: compute-change-detection-coverage
+    run: |
+      # Compute whether to use change detection and coverage
+      import json
+      import os
+      import pathlib
+      import sys
+
+      FILE_APPEND_MODE = 'a'
+      OUTPUTS_FILE_PATH = pathlib.Path(os.environ['GITHUB_OUTPUT'])
+
+      def set_output(name, value):
+          with OUTPUTS_FILE_PATH.open(FILE_APPEND_MODE) as outputs_file:
+              outputs_file.writelines(f'{name}={value}{os.linesep}')
+
+      # Input from GHA
+      pull_request_change_detection = json.loads('${{
+        inputs.pull-request-change-detection
+      }}')
+      pull_request_branch = '${{ github.event.pull_request.base.ref || '' }}'
+
+      # Compute coverage and change detection arguments
+      coverage_arg = '--coverage'
+      change_detection_arg = ''
+      if pull_request_branch and pull_request_change_detection:
+          coverage_arg = ''
+          change_detection_arg = (
+              f'--changed --base-branch {pull_request_branch}'
```

Comment:
Does this work with all of `integration`, `sanity` and `units` subcommands?

Reply:
Yes. (https://github.com/felixfontein/ansible-playground/pull/4 uses it for all.)